src/admin/class-datalayers.php

```diff
@@ -539,31 +744,92 @@ function remove_required($row) {
 	}
 
 	/**
-	 * Get the source fields from the datalayer file.
+	 * Fetch the data from an external source URL.
+	 *
+	 * @return mixed
+	 */
+	public static function fetch_datalayer_url_data() {
+		$url      = self::$datalayer_url;
+		$response = wp_remote_get( $url );
+
+		if ( is_wp_error( $response ) ) {
+			return [];
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+
+		if ( ! $body ) {
+			return [];
+		}
+
+		return $body;
+	}
+
+	/**
+	 * Get the source fields from the datalayer URL.
 	 *
 	 * @param int|string $object_id The object ID.
 	 *
 	 * @return array
 	 */
-	public static function cmb2_get_source_fields( $object_id ) {
-		// Get source fields from the datalayer file.
-		$datalayer_file = get_post_meta( $object_id, 'datalayer_file_id', true );
-		if ( ! $datalayer_file ) {
-			return [];
+	public static function get_datalayer_source_fields( $object_id ) {
+		// First try to retrieve the source fields from the postmeta.
+		$source_fields = get_post_meta( $object_id, 'source_fields', true );
+
+		if ( ! empty( $source_fields ) ) {
+			// Get all field labels from the source_fields array.
+			$source_fields = array_map(
+				function ( $field ) {
+					return $field['field_label'];
+				},
+				$source_fields
+			);
+
+			return $source_fields;
 		}
 
-		$file = get_attached_file( $datalayer_file );
-		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- file_get_contents is allowed here.
-		$data = file_get_contents( $file );
-		$geom = geoPHP::load( $data );
-		$json = $geom->out( 'json' );
-		$data = json_decode( $json, true );
+		// If no source fields are set, get the source fields from the datalayer file.
+		switch ( self::$datalayer_type ) {
+			case 'fileinput':
+			default:
+				$file = get_attached_file( get_post_meta( $object_id, 'datalayer_file_id', true ) );
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- We need to read the file contents.
+				$data = file_get_contents( $file );
+
+				break;
+			case 'url':
+				$data = self::fetch_datalayer_url_data( $object_id );
+
+				break;
+		}
 
 		$source_fields = [];
-		if ( $data ) {
-			foreach ( $data['features'][0]['properties'] as $key => $val ) {
-				$source_fields[] = $key;
+
+		// Check if data is valid GeoJSON and if we can parse it. If not, a IOException will be thrown.
+		try {
+			$geom = geoPHP::load( $data );
+
+			if ( ! $geom->getComponents() && ! $geom->getComponents()[0] ) {
+				return [];
+			}
+
+			if ( $geom->getComponents()[0]->getData() ) {
+				foreach ( $geom->getComponents()[0]->getData() as $key => $val ) {
```

Comment:
Zou het voor kunnen komen dat een veld niet in de eerste component zit, maar wel in een latere? Geen idee hoor... Aan de andere kant wil je liever ook niet alle componenten doorlopen als dat niet nodig is.

Reply:
In theorie kan dat natuurlijk. Ik weet niet of het in de praktijk voor gaat komen. Ik stel voor dat we het voor nu zo laten en als we een use case tegenkomen waarin blijkt dat dit voorkomt, dan kunnen we het nog aanpassen.